src/testSession.ts

```diff
@@ -235,55 +245,83 @@ export class TestSession {
-          cmd += ' --json';
-        }
+          // Add the json flag if it looks like an sfdx command so we can return
+          // parsed json in the command return.
+          if (cmd.split(' ')[0].includes('sfdx') && !cmd.includes('--json')) {
+            cmd += ' --json';
+          }
 
-        const rv = shell.exec(cmd, { silent: true });
-        if (rv.code !== 0) {
-          const io = cmd.includes('--json') ? rv.stdout : rv.stderr;
-          throw Error(`Setup command ${cmd} failed due to: ${io}`);
-        }
-        dbug(`Output for setup cmd ${cmd} is:\n${rv.stdout}`);
-
-        // Automatically parse json results
-        if (cmd.includes('--json')) {
-          try {
-            const jsonOutput = parseJson(rv.stdout);
-            // keep track of all org creates
-            if (cmd.includes('org:create')) {
-              const username = getString(jsonOutput, 'result.username');
-              if (username) {
-                dbug(`Saving org username: ${username} from ${cmd}`);
-                this.orgs.push(username);
+          const rv = shell.exec(cmd, { silent: true });
+          // const rv = this.execWithRetries(cmd);
+          if (rv.code !== 0) {
+            const io = cmd.includes('--json') ? rv.stdout : rv.stderr;
+            throw Error(`Setup command ${cmd} failed due to: ${io}`);
+          }
+          dbug(`Output for setup cmd ${cmd} is:\n${rv.stdout}`);
+
+          // Automatically parse json results
+          if (cmd.includes('--json')) {
+            try {
+              const jsonOutput = parseJson(rv.stdout);
+              // keep track of all org creates
+              if (cmd.includes('org:create')) {
+                const username = getString(jsonOutput, 'result.username');
+                if (username) {
+                  dbug(`Saving org username: ${username} from ${cmd}`);
+                  this.orgs.push(username);
+                }
               }
+              this.setup.push(jsonOutput);
+            } catch (err: unknown) {
+              dbug(`Failed command output JSON parsing due to:\n${(err as Error).message}`);
+              this.setup.push(rv);
             }
-            this.setup.push(jsonOutput);
-          } catch (err: unknown) {
-            dbug(`Failed command output JSON parsing due to:\n${(err as Error).message}`);
+          } else {
             this.setup.push(rv);
           }
-        } else {
-          this.setup.push(rv);
         }
       }
+    };
+
+    let attempts = 0;
+    let completed = false;
+    const timeout = new Duration(env.getNumber('TESTKIT_SETUP_RETRIES_TIMEOUT') ?? 5000, Duration.Unit.MILLISECONDS);
+
+    while (!completed && attempts <= this.setupRetries) {
+      try {
+        dbug(`Executing setup commands (attempt ${attempts + 1} of ${this.setupRetries + 1})`);
+        setup();
+        completed = true;
+      } catch (err) {
+        attempts += 1;
```

Comment:
scenario: scratch org created successfully but something else in setupCommands failed.

Do we need/want to delete the org?  Or do we expect that the re-run will overwrite anything important (aliases, usernames, default-ness, etc) and then it'll get cleaned up at the end?